tests/unit/core/reducers/test_addons.js

```diff
@@ -789,4 +781,38 @@ describe(__filename, () => {
       expect(state.loadingBySlug).toEqual({ [slug2]: false });
     });
   });
+
+  describe('createPlatformFiles', () => {
+    it('creates a default object if there is no version', () => {
+      expect(createPlatformFiles(undefined)).toEqual(defaultPlatformFiles);
+    });
+
+    it('creates a default object if there are no files', () => {
+      expect(createPlatformFiles({ ...fakeVersion, files: [] })).toEqual(
+        defaultPlatformFiles,
+      );
+    });
+
+    it('creates a PlatformFilesType object from a version with files', () => {
+      const windowsFile = {
+        ...fakePlatformFile,
+        platform: OS_WINDOWS,
+      };
+      const unknownPlatform = 'unknownPlatform';
+      const unknownFile = {
+        ...fakePlatformFile,
+        platform: unknownPlatform,
+      };
+      expect(
+        createPlatformFiles({
+          ...fakeVersion,
+          files: [windowsFile, unknownFile],
+        }),
+      ).toEqual({
+        ...defaultPlatformFiles,
+        [OS_WINDOWS]: windowsFile,
+        [unknownPlatform]: unknownFile,
```

Comment:
This is a special case and I think it warrants its own test. Also, you can delete [this one](https://github.com/mozilla/addons-frontend/blob/159016d1a7bbbad8a997ea978d3accce1f6b9095/tests/unit/core/reducers/test_addons.js#L263) because it makes more sense as a unit test of `createPlatformFiles` now.